src/steps/generic.rs

```diff
@@ -590,14 +590,30 @@ pub fn run_pixi_update(ctx: &ExecutionContext) -> Result<()> {
 
     // Check if `pixi --help` mentions self-update, if yes, self-update must be enabled.
     // pixi self-update --help works regardless of whether the feature is enabled.
-    let output = ctx.run_type().execute(&pixi).arg("--help").output_checked()?;
+    let top_level_help_output = ctx.run_type().execute(&pixi).arg("--help").output_checked()?;
```

Comment:
Not your code, but let's fix this
```suggestion
    let top_level_help_output = ctx.run_type().execute(&pixi).arg("--help").output_checked_utf8()?;
```